src/common.ml

```diff
@@ -20,17 +20,20 @@ type token =
 
 (* Constants *)
 type const =
-      AtomConst of string    (* atoms    *)
     | IntConst of int        (* integers *)
     | FloatConst of float    (* floats   *)
     | StringConst of string  (* strings  *)
     | BoolConst of bool      (* booleans *)
 
 (* Expressions *)
 type exp =
-      VarExp of string              (* variables                *)
+    | VarExp of string              (* variables                *)
     | ConstExp of const             (* constants                *)
-    | RuleExp of exp * exp          (* Head :- Body.            *)
-    | FactExp of exp                (* Head.                    *)
-    | CompoundTermExp of exp * exp  (* functor(arg1, arg2, ...) *)
-    | ListExp of exp * exp          (* arg1, arg2               *)
+    | TermExp of string * exp list  (* functor(arg1, arg2, ...) *)
+    | ConjunctionExp of exp * exp   (* term, term               *)
+    | DisjunctionExp of exp * exp   (* term; term               *)
+
+(* Declarations *)
+type dec =
+    | Clause of exp * exp   (* Head :- Body. *)
```

Comment:
Facts can be represented like so:
```
Clause (
    TermExp (...),
    ConstExp (BoolConst true)
)
```
In this sense, `Head.` is syntactic sugar for `Head :- true.`